packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.ts

```diff
@@ -0,0 +1,198 @@
+import { HttpStatus, Injectable, Logger } from '@nestjs/common';
+import {
+  CreateQualificationDto,
+  AssignQualificationDto,
+  UnassignQualificationDto,
+  QualificationDto,
+} from './qualification.dto';
+import { QualificationEntity } from './qualification.entity';
+import { ErrorQualification } from '../../common/constants/errors';
+import { ControlledError } from '../../common/errors/controlled';
+import { QualificationRepository } from './qualification.repository';
+import { UserEntity } from '../user/user.entity';
+import { UserRepository } from '../user/user.repository';
+import { UserStatus, Role } from '../../common/enums/user';
+import { UserQualificationEntity } from './user-qualification.entity';
+import { ServerConfigService } from '../../common/config/server-config.service';
+
+@Injectable()
+export class QualificationService {
+  private readonly logger = new Logger(QualificationService.name);
+
+  constructor(
+    private readonly qualificationRepository: QualificationRepository,
+    private readonly userRepository: UserRepository,
+    private readonly serverConfigService: ServerConfigService,
+  ) {}
+
+  public async createQualification(
+    createQualificationDto: CreateQualificationDto,
+  ): Promise<QualificationDto> {
+    const newQualification = new QualificationEntity();
+    newQualification.reference = createQualificationDto.reference;
+    newQualification.title = createQualificationDto.title;
+    newQualification.description = createQualificationDto.description;
+
+    if (createQualificationDto.expiresAt) {
+      const providedExpiresAt = new Date(createQualificationDto.expiresAt);
+      const now = new Date();
+      const minimumValidUntil = new Date(
+        now.getTime() +
+          this.serverConfigService.qualificationMinValidity * 1000,
+      );
+
+      if (providedExpiresAt <= minimumValidUntil) {
+        const minValidityHours =
+          this.serverConfigService.qualificationMinValidity / 3600;
+        const errorMessage = ErrorQualification.InvalidExpiresAt.replace(
```

Comment:
The message says hours. But qualificationMinValidity seems to be seconds, am I right?